subprojects/logging/src/main/java/org/gradle/internal/logging/console/ThrottlingOutputEventListener.java

```diff
@@ -26,27 +26,38 @@
 import java.util.List;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
 /**
  * Queue output events to be forwarded and schedule flush when time passed or if end of build is signalled.
  */
 public class ThrottlingOutputEventListener implements OutputEventListener {
-    private final BatchOutputEventListener listener;
+    private final OutputEventListener listener;
 
     private final ScheduledExecutorService executor;
     private final TimeProvider timeProvider;
     private final int throttleMs;
     private final Object lock = new Object();
 
+    private final Runnable scheduledRenderNow = new Runnable() {
```

Comment:
It feels a bit odd to add this here instead of managing this in the `BuildStatusRenderer`. It seems like we could reduce the amount of code churn decently if we move it there. Is there a technical reason for this or was this in prep to add timers to WIP renderer as well?